packages/mui-material/src/Backdrop/Backdrop.js

```diff
@@ -69,8 +70,20 @@ const Backdrop = React.forwardRef(function Backdrop(inProps, ref) {
 
   const rootSlotProps = slotProps.root ?? componentsProps.root;
 
+  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };
+
+  const [TransitionSlot, transitionProps] = useSlot('transition', {
+    elementType: Fade,
+    externalForwardedProps: {
+      slots: backwardCompatibleSlots,
+    },
+    ownerState,
+  });
```

Comment:
This is where I've seen other components pass the transition props. Is this missing from the Component?

Reply:
No need to add `TransitionProps`, as it didn't exist before, but we should provide the slotProps:

```suggestion
  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };

  const [TransitionSlot, transitionProps] = useSlot('transition', {
    elementType: Fade,
    externalForwardedProps: {
      slots: backwardCompatibleSlots,
      slotProps,
    },
    ownerState,
  });
```